lib/hocon/impl/config_delayed_merge.rb

```diff
@@ -35,10 +46,189 @@ def unwrapped
     raise Hocon::ConfigError::ConfigNotResolvedError.new(error_message, nil)
   end
 
+  def resolve_substitution(context, source)
+    self.class.resolve_substitution(self, stack, context, source)
+  end
+
+  def self.resolve_substitution(context, source, replaceable = self, stack = @stack)
```

Comment:
I think these are called `resolveSubstitutions` in the upstream, since more than one substitution can be resolved.